File: Remove All Occurrences of a Substring/ans.cpp
class Solution {
public:
    string removeOccurrences(string s, string part) {

        while (true) {
            int i = s.find(part);
            if (i == string::npos) {
                return s;
            } else {
                s.erase(i, part.size());
            }
            
        }
   
    }
};

File: Remove All Occurrences of a Substring/ans.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        p = len(part)
        part_arr = [c for c in part]
        stack = []

        for c in s:
            stack.append(c)
            while len(stack) >= p and stack[-p:] == part_arr:
                stack = stack[:-p]
        
        return "".join(stack)


        
